sashimi-webapp/src/database/create/entitiesCreator.js

```diff
@@ -1,22 +1,11 @@
-import constants from '../constants';
-import tableCreator from './tableCreator';
-import exceptions from '../exceptions';
-import initDataGenerator from './initDataGenerator';
-import SqlCommands from '../sql-related/sqlCommands';
+import constants from 'src/database/constants';
+import tableCreator from 'src/database/create/tableCreator';
+import exceptions from 'src/database/exceptions';
+import initDataGenerator from 'src/database/create/initDataGenerator';
+import SqlCommands from 'src/database/sql-related/sqlCommands';
 
 const sqlCommands = new SqlCommands();
 
-// a promise to allow sequential running of keying in data
-function initFillUpDefaultDatas() {
-  if (typeof Promise === 'function') {
-    return new Promise((resolve, reject) => {
-      resolve(true);
-    });
-  } else {
-    throw new exceptions.PromiseFunctionNotDefined();
-  }
-}
-
 function isTableExists(tableName) {
   if (typeof Promise === 'function') {
```

Comment:
Related to this: https://github.com/nus-mtp/lecture-note-2.0/pull/241#pullrequestreview-25959209

Let's continue the conversation here: 

You mentioned that
> I will not be able to know when the developer first use or import my library >.< because if I put a check in the constructor, it will only be called the if user use const something = new Database() or something. The caller can basically be able to call any of the functions directly

Correct me if I am wrong, I think your concern was that, 
a developer can call your other functions without calling `const something = new Database()` himself.
Example:
`main.js` call `new Database()`
`FileManager` call `storage.loadFile()`

Thus if you put the check for Promise in `new Database()`, 
then `storage.loadFile()` won't be able to check if Promise exist before it is called?

If that was your concern, then here is some explanations to why the aggressive checking is not needed.

### `Promise` will be transpiled to ES5 code to support browser that does not support `Promise`

Promise is a native function in ES6 (which is what we are writing now). 
Our ES6 codes will be transpiled to ES5 codes using Babel and [Babel polyfill](https://babeljs.io/docs/usage/polyfill/) (which will allow us to use Promise in JavaScript engine that does not yet support ES6).
All the codes inside the `src` folder will the transpiled into ES5 syntax.

This will imply that, everyone will have access to the Promise as long as the code is placed inside the `src` folder.
Thus, using the previous example:
`main.js` call `new Database()` <-- This will have access to Promise
`FileManager` call `storage.loadFile()` <-- This will also have access to Promise

### The JavaScript engine will already throw the exception if `Promise` is not supported

Secondly, let's assume that, somehow the database code is being executed from a ES5 JavaScript engine (where Promise is not natively supported yet). 

The Promise object will be `undefined` by default. 
Thus, the JavaScript engine/the browser will throw an error indicating that `'Promise' is undefined.`. 
Which is doing essentially what you are doing already `throw new exceptions.PromiseFunctionNotDefined();`